src/models/transaction_receipt.rs

```diff
@@ -65,24 +64,7 @@ impl StarknetTransactionReceipt {
                                 // If to is Some, means contract_address should be None as it is a normal transaction
                                 Some(_) => None,
                                 // If to is None, is a contract creation transaction so contract_address should be Some
-                                None => {
-                                    let event = events
-                                        .iter()
-                                        .find(|event| event.keys.iter().any(|key| *key == EVM_CONTRACT_DEPLOYED))
-                                        .ok_or(EthApiError::Other(anyhow::anyhow!(
-                                    "Kakarot Core: No contract deployment event found in Kakarot transaction receipt"
-                                )))?;
-
-                                    let evm_address =
-                                        event.data.first().ok_or(DataDecodingError::InvalidReturnArrayLength {
-                                            entrypoint: "deployment".into(),
-                                            expected: 1,
-                                            actual: 0,
-                                        })?;
-
-                                    let evm_address = Felt252Wrapper::from(*evm_address);
-                                    Some(evm_address.try_into()?)
-                                }
+                                None => Some(create_address(eth_tx.from, eth_tx.nonce.as_u64())),
```

Comment:
Can we add an invariant for now?
For instance, check that there is a proxy at the get_class_hash_at(compute_starknet_address(evm_address)) == PROXY_CLASS_HASH

Reply:
but this will slow down the execution because we have to make an extra call a provider :/

Reply:
but in the early days we need to make sure that there is a contract deployed there, no?
what is the "fastest" way to check this invariant?

Reply:
Alright!

Reply:
I think the only way is indeed to call the provider. But would prefer to avoid this.